File: cs12b/lab2/FileReverse.java
// FileReverse.java
// Reverse strings from one file to another
// Alec Felt  1430374  allfelt@ucsc.edu
import java.io.*;
import java.util.Scanner;
class FileReverse{	

	public static String stringReverse(char[] temp, String s, int n) {
		
		if (n>0) {
			temp[n-1] = s.charAt(s.length() - n);
			stringReverse(temp, s, n-1);
		}
		String sk = new String(temp);
		return sk;
	}

	public static void main(String[] args) throws IOException{
 
		Scanner in = null;
		PrintWriter out = null;
		String line = null;
		String[] token = null;
		int i, n, lineNumber = 0;
	
		// check number of command line arguments is at least 2
		if(args.length < 2){
			System.out.println("Usage: FileCopy <input file> <output file>");
			System.exit(1);
		}
	
		// open files
		in = new Scanner(new File(args[0]));
		out = new PrintWriter(new FileWriter(args[1]));
	
		// read lines from in, extract and print tokens from each line
		while( in.hasNextLine() ){
			lineNumber++;
	
			// trim leading and trailing spaces, then add one trailing space so
			// split works on blank lines
			line = in.nextLine().trim() + " ";
 
			// split line around white space
			token = line.split("\\s+");
 
			// print out tokens
			n = token.length;
			char[] temp;
			for(i=0; i<n; i++){
				temp = new char[token[i].length()]; 
				out.println(""+stringReverse(temp, token[i], token[i].length()));
			}
		}
		// close files
		in.close();
		out.close();
	}
}

File: cs12b/lab3/FileReverse.c
//Alec Felt  1430374  allfelt@ucsc.edu

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void stringReverse(char* s) {
	char temp;
	int i = 0;
	int j = strlen(s) - 1;

	while (i < j) {
		temp = s[i];
		s[i] = s[j];
		s[j] = temp;
		i++;
		j--;
	}	
}

int main(int argc, char* argv[]) {
	
	FILE* in;
	FILE* out;
	char words[256];
	
	if (argc != 3) {
		printf("Usage: %s <input file> <output file>\n", argv[0]);
		exit(EXIT_FAILURE);
	}

	in = fopen(argv[1], "r");
	if (in == NULL) {
		printf("Unable to read from file: %s ", argv[1]);
		exit(EXIT_FAILURE);
	}

	out = fopen(argv[2], "w");
	if (out == NULL) {
		printf("Unable to write to file: %s", argv[2]);
		exit(EXIT_FAILURE);
	}

	while (fscanf(in, " %s", words) != EOF) {
		stringReverse(words);
		fprintf(out, "%s\n", words);
	}

	fclose(in);
	fclose(out);
	
	return(EXIT_SUCCESS);
}
